tools/cmd.go

```diff
@@ -67,3 +70,30 @@ func mustGetDuration(cmd *cobra.Command, flagName string) time.Duration {
 	}
 	return val
 }
+
+func ResolveManifestFile(input string) (manifestName string, err error) {
+	if input == "" {
+		if _, err := os.Stat("substreams.yaml"); err == nil {
+			return "substreams.yaml", nil
+		} else if errors.Is(err, os.ErrNotExist) {
+			return "", fmt.Errorf("finding manifest file: %w", err)
+		}
+	} else {
+
+		// Check valid file first
+		if _, err := os.Stat(input); err == nil {
+			_, err := os.Open(input)
+			if err == nil {
+				return input, nil
+			}
```

Comment:
You should first investigate the method you are using, for example `os.Stat` to see if something could be useful. After that, I also suggest that you perform some Google search for the specific problem your are tackling.

Opening a path to check what kind it is is clearly not correct.